### DISCUSSION TITLE
aubuf with smart maximum

### DISCUSSION THREAD

**Discussion by cspiel1** [baresip/rem#16] [Ideas]
The Discussion feature https://github.com/baresip/re/discussions is very handy. Should we also enable this for rem?

This is not an issue, more a discussion about `aubuf`.

The idea is to have an audio buffer that have:
- min frame size
- max frame size
- smart max frame size, with silence detection (supporting the multiple sample formats)

If a smart max size is specified (smaller than the max size) and an upper boundary for the sample value, then a frame is dropped only if it is a silence frame.

About the solution:
- `aubuf` seems only buffer byte arrays
- maybe audio processing should be converted to track `auframe` queues

**Comment by sreimers** [comment]
This PR adds aubuf auframe support https://github.com/baresip/rem/pull/7
It still needs to be better tested, but should be almost ready.

**Reply by cspiel1** [reply]
So now this would make it possible to add a silence detection for all supported sample formats, right?

**Reply by sreimers** [reply]
Yes that should be possible. This week I am a bit busy due to other project deadlines. Will test and merge the PR next week.
